### PR TITLE
feat: imagePullSecrets for servers and controllers

### PR DESCRIPTION
**Pull request by hobti01**
* Added imagePullSecrets
* Added annotations to argo-events CRDs to deploy with Helm 3 to allow testing
* Added dependent charts to gitignore

Checklist:

* [ x ] I have update the chart version in `Chart.yaml` following Semantic Versioning.
* [ x ] Any new values are backwards compatible and/or have sensible default.
* [ x ] I have followed the testing instructions in the [contributing guide](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md).
* [ x ] I have signed the CLA and the build is green.
* [ x ] I will test my changes again once merged to master and published.

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

**Comment by spencergilbert** [comment]
I usually don't see PRs hitting all of the charts like this - I'm happy to approve the argo-cd changes but will defer to the other owners for their charts

File: charts/argo/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 appVersion: v2.8.0
 description: A Helm chart for Argo Workflows
 name: argo
-version: 0.9.4
+version: 0.10.0
```
Comment (stefansedich) [inline charts/argo/Chart.yaml:5]: We normally just bump the patch in the argo chart and bump the minor when pushing a new version (i.e. the argo image version is upgraded)

Comment (hobti01) [reply charts/argo/Chart.yaml:5]: OK, we usually bump our minor versions when changing the UX - i.e. new properties.  Your guidelines say a minor version for new chart functionality - but as you like.  Please let me know if you want this changed or you're just sharing information :)

**Comment by hobti01** [comment]
> I usually don't see PRs hitting all of the charts like this - I'm happy to approve the argo-cd changes but will defer to the other owners for their charts

@spencergilbert I wasn't sure how this repo is separating things since it's all together.  I see there are already multiple implementations of the same functionality in the charts so I thought to get in front of that.  Sorry if this is causing friction.

**Comment by hobti01** [comment]
Hi all - going on a month for what I thought was a small change.  Is there a better way for me to help move this forward?

**Comment by spencergilbert** [comment]
I guess this is pretty well approved.. resolve the conflicts and we can merge. If none of the other maintainers have commented/approved yet I doubt they will or mind the change.

**Comment by hobti01** [comment]
Thanks @spencergilbert , I've rebased and taken the advice of @stefansedich for patch versions.

**Comment by spencergilbert** [comment]
Sorry, it looks like this got forgotten and has a number of conflicts now

**Comment by hobti01** [comment]
@spencergilbert I have rebased and updated the new workloads for imagePullSecrets, thank you for your update

**Comment by spencergilbert** [comment]
Thanks, sorry for the long wait

**Pull request merged**